crates/aggchain-proof-builder/src/lib.rs

```diff
@@ -267,17 +267,55 @@ impl<ContractsClient> AggchainProofBuilder<ContractsClient> {
             .map(|inserted_ger| inserted_ger.ger())
             .collect();
 
-        // NOTE: Corresponds to all of them because we do not have unset claims yet.
-        let bridge_exits_claimed: Vec<GlobalIndexWithLeafHash> = request
-            .aggchain_proof_inputs
-            .imported_bridge_exits
-            .iter()
-            .filter(|ib| new_blocks_range.contains(&ib.block_number))
-            .map(|ib| GlobalIndexWithLeafHash {
+        // Helper function to filter, sort, and convert a vector of items to the desired
+        // output type. It filters items based on a range of blocks, sorts them
+        // using type Ord implementation, and maps them to a new type using the
+        // provided mapping function.
+        fn filter_sort_map<T, F, U>(
+            items: Vec<T>,
+            range: &std::ops::RangeInclusive<u64>,
+            block_number_fn: fn(&T) -> u64,
+            map_fn: F,
+        ) -> Vec<U>
+        where
+            F: Fn(T) -> U,
+            T: Ord,
+        {
+            let mut filtered_items: Vec<_> = items
+                .into_iter()
+                .filter(|item| range.contains(&block_number_fn(item)))
+                .collect();
+            filtered_items.sort();
+            filtered_items.into_iter().map(map_fn).collect()
+        }
+
+        // NOTE: Corresponds to all the bridge exits claimed because we do not have
+        // unset claims yet.
+        let bridge_exits_claimed: Vec<GlobalIndexWithLeafHash> = filter_sort_map(
+            request.aggchain_proof_inputs.imported_bridge_exits,
+            &new_blocks_range,
+            |ib| ib.block_number,
+            |ib| GlobalIndexWithLeafHash {
                 global_index: ib.global_index.into(),
                 bridge_exit_hash: ib.bridge_exit_hash.0,
-            })
-            .collect();
+            },
+        );
+
+        // Prepare removed GERS for the proof.
+        let removed_gers: Vec<Digest> = filter_sort_map(
+            request.aggchain_proof_inputs.removed_gers,
+            &new_blocks_range,
+            |removed_ger| removed_ger.block_number,
+            |removed_ger| removed_ger.global_exit_root,
+        );
+
+        // Prepare unset claims for the proof.
+        let unset_claims: Vec<Digest> = filter_sort_map(
```

Comment:
I think we have an error here, as @hadjiszs pointed out before, something wasn't ok.

  ```
unsetGlobalIndexHashChain = Hashes.efficientKeccak256(
                unsetGlobalIndexHashChain,
                bytes32(globalIndex)
            );
```
The `unset_claims` are a vector of `globalIndex` indicating uniquely which leafs were unclaimed
```
     * @param globalIndexes Global index is defined as:
     * | 191 bits |    1 bit     |   32 bits   |     32 bits    |
     * |    0     |  mainnetFlag | rollupIndex | localRootIndex |
```

¡